### DISCUSSION TITLE
Render or Load .vtr file using VEDO

### DISCUSSION THREAD

**Discussion by prithishh3** [marcomusy/vedo#303] [Q&A] [answered]
Hi,

I have two questions:

1. Is it possible to render or load .VTR file using VEDO ?
2. How to import x,y,z data in vedo and create surface rendered or mesh rendered 3D visualization ?

**Comment by marcomusy** [comment, marked as answer]
Hi Prithish, 

1. yes but it's very much limited to visualization as `load()` only creates the polygonal mesh of the geometry of the volume, e.g.:
```python
from vedo import *
rg = load('https://www.dropbox.com/s/ib96jr4kknvvrcl/RectilinearGrid.vtr')
rg.lineWidth(0.1)
rg.printInfo()
show(rg, axes=1)
```
![image](https://user-images.githubusercontent.com/32848391/106045382-259ebc00-60e1-11eb-88bf-c09197d8c7fd.png)
if you need to do more complicated analysis I suggest to have a look at [pyvista](https://github.com/pyvista/pyvista) which naturally supports this type of data set.

2. Yes - there is an example on how to reconstruct a polygonal mesh from a point cloud: https://vedo.embl.es/#gallery (search "reco")
You can `load()` a point cloud from a `.xyz` ascii file and/or from command line, eg.
`vedo https://vedo.embl.es/examples/data/teapot.xyz`
![image](https://user-images.githubusercontent.com/32848391/106047196-97780500-60e3-11eb-9b26-3ad1654a633f.png)

**Reply by prithishh3** [reply]
Thank you for the assistance.

I followed your suggestion regarding pyvista and my problem regarding .vtr file is solved.

I have used filters like contour() to render the surface and then generated the mesh. Is there any similar function in "vedo" ?
```
mesh = pv.read('tcombine_1.vtr')
contours = mesh.contour()

```
This is how I used the contour() filter in pyvista

**Reply by marcomusy** [reply]
Hi - thanks for the feedback - no, at the minute there's no support in `vedo` for rectilinear grid dataset.. maybe for future developments :)
